tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8013,6 +8013,54 @@ def err_hlsl_reorder_unsupported_stage : Error<
    "dx::MaybeReorderThread is unavailable in shader stage '%0' (requires 'raygeneration')">;
 def err_hlsl_hitobject_unsupported_stage : Error<
    "dx::HitObject is unavailable in shader stage '%0' (requires 'raygeneration', 'closesthit' or 'miss')">;
+
+// Linear Algebra Operations
+def err_hlsl_linalg_param_must_be_const : Error<
+  "'%0' must be a constant parameter">;
+def err_hlsl_linalg_vector_incorrect_type : Error<
+  "%0 is incorrect type, must be 16-bit or 32-bit 'unsigned int', "
+  "'signed int' or 'float'">;
+def err_hlsl_linalg_isunsigned_incorrect_for_given_type : Error<
+  "%0 must be %select{false|true}1 for a %2 vector type">;
+def err_hlsl_linalg_interpretation_value_incorrect : Error<
+  "%0 is an invalid %select{Memory|Register}1 Interpretation value">;
+def err_hlsl_linalg_matrix_layout_is_not_transposable : Error<
+  "RowMajor and ColumnMajor matrices are not transposable">;
+def err_hlsl_linalg_optimal_matrix_layout_matrix_stride_must_be_zero : Error<
+  "for optimal matrix layout, matrix stride must be zero">;
```

Comment:
```suggestion
  "for optimal matrix layout, matrix stride must be greater than 0">;
```

Reply:
this one must be zero. (it's a requirement for optimal layouts)

Reply:
Doh. I misread it, but we should use `0` not `zero` to avoid localization issues.